src/util/http_utils.js

```diff
@@ -946,6 +946,55 @@ function set_response_headers_from_request(req, res) {
     if (req.query['response-expires']) res.setHeader('Expires', req.query['response-expires']);
 }
 
+/**
+ * Authenticate JWT bearer token for metrics / version endpoints.
+ * Returns `true` on success, `false` after the function already sent an HTTP
+ * response (401/403) and the caller should terminate the handler early.
+ * 
+ * @param {import('http').IncomingMessage} req
+ * @param {import('http').ServerResponse} res
+ * @param {string[]} roles
+ */
+function authorize_bearer(req, res, roles = undefined) {
+    const { authorization, ...rest_headers } = req.headers;
+    if (!authorization) {
+        dbg.error('Authentication required:', req.method, req.url, rest_headers);
+        // request lacks authentication, let the client know it's required with 401 Unauthorized
+        res.statusCode = 401;
+        res.setHeader('WWW-Authenticate', 'Bearer');
+        res.setHeader('Content-Type', 'text/plain');
+        res.end('Unauthorized');
```

Comment:
Maybe just create a function in the scope of this function (a const function) and call it all the three places to avoid dups.